x-pack/plugins/task_manager/server/ephemeral_task_lifecycle.ts

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Subject, Observable } from 'rxjs';
+import { filter } from 'rxjs/operators';
+import { Logger } from '../../../../src/core/server';
+
+import { Result, asErr, asOk } from './lib/result_type';
+import { TaskManagerConfig } from './config';
+
+import { asEphemeralTaskDelayedDueToCapacityEvent, TaskEventType } from './task_events';
+import { Middleware } from './lib/middleware';
+import { EphemeralTaskInstance } from './task';
+import { TaskTypeDictionary } from './task_type_dictionary';
+import { TaskLifecycleEvent } from './polling_lifecycle';
+import { EphemeralTaskManagerRunner } from './task_running/ephemeral_task_runner';
+import { TaskPool } from './task_pool';
+
+export interface EphemeralTaskLifecycleOpts {
+  logger: Logger;
+  definitions: TaskTypeDictionary;
+  config: TaskManagerConfig;
+  middleware: Middleware;
+  elasticsearchAndSOAvailability$: Observable<boolean>;
+  pool: TaskPool;
+  lifecycleEvent: Observable<TaskLifecycleEvent>;
+}
+
+export type EphemeralTaskInstanceRequest = Omit<EphemeralTaskInstance, 'startedAt'>;
+
+/**
+ * The public interface into the task manager system.
+ */
+export class EphemeralTaskLifecycle {
```

Comment:
We should add unit tests around this.
I didn't do that when I spiked this as I expected you to throw it out and start from scratch ;)
This is now doing lots of things and should definitely be validated in unit tests.